### DISCUSSION TITLE
Solver Missing Hints When Reading CCS

### DISCUSSION THREAD

**Discussion by BeratOz01** [Consensys/gnark#604] [Q&A] [answered]
Hello!
I have Modulus hint function
```
func Modulus(_ *big.Int, inputs []*big.Int, result []*big.Int) error {
	result[0].Mod(inputs[0], inputs[1])
	return nil
}
```
I registers Modulus hint function when compiling circuit.
```
hint.Register(Modulus)
```
But when I try to read ccs from file, I am encountering this error `solver missing hint(s):`. How can I solve this problem ?

Thank you!!

Edit:
When I call `hint.Name(Modulus)`, function returns `main.Modulus`. I think that function is already registered but still same getting same error.

**Comment by BeratOz01** [comment]
Hello again!
I made progression on this error but still can not figure out how to use hint function outside of main package. My circuit will not be in the main package, but every time it searches for the Modulus hint function, it looks at the main.Modulus, so it cannot find it.

**Reply by ivokub** [reply, marked as answer]
Hi! You can use the [WithHints](https://pkg.go.dev/github.com/consensys/gnark@v0.8.0/backend#WithHints) option to the prover to provide the hint to the prover.
